File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'category_id',
        'name',
        'description',
        'price',
        'image',
        'skin_type',
        'key_ingredients',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }
}

File: lib/data/models/product.dart
class Product {
  final String id;
  final String category;
  final String name;
  final double price;
  final double rating;
  final String imageAsset;
  final String description;
  final String skinType;
  final String keyIngredients;

  const Product({
    required this.id,
    required this.category,
    required this.name,
    required this.price,
    required this.rating,
    required this.imageAsset,
    required this.description,
    required this.skinType,
    required this.keyIngredients,
  });
}